cluster/examples/kubernetes/ceph/csi/rbd/storageclass-ec.yaml

```diff
@@ -0,0 +1,70 @@
+#################################################################################################################
+# Create a storage class with a data pool that uses erasure coding for a production environment.
+# A metadata pool is created with replication enabled. A minimum of 3 nodes with OSDs are required in this
+# example since the default failureDomain is host.
+#  kubectl create -f storageclass-ec.yaml
+#################################################################################################################
+
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: replicated-metadata-pool
+  namespace: rook-ceph
+spec:
+  replicated:
+    size: 2
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+  name: ec-data-pool
+  namespace: rook-ceph
+spec:
+  # Make sure you have enough nodes and OSDs running bluestore to support the replica size or erasure code chunks.
+  # For the below settings, you need at least 3 OSDs on different nodes (because the `failureDomain` is `host` by default).
+  erasureCoded:
+    dataChunks: 2
+    codingChunks: 1
+---
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+   name: rook-ceph-block
+provisioner: rook-ceph.rbd.csi.ceph.com
+parameters:
+    # clusterID is the namespace where the rook cluster is running
+    # If you change this namespace, also change the namespace below where the secret namespaces are defined
```

Comment:
This seems like a poor user experience. Why doesn't rook internally use `clusterID` to inform its knowledge of `csi.storage.k8s.io/provisioner-secret-namespace` and `csi.storage.k8s.io/node-stage-secret-namespace`?

Reply:
Agreed this is painful, but the CSI driver has been like this from the start and we had discussions a while back about how it could be improved. With the CSI spec requiring the other properties below, I haven't seen how we can simplify this.

Reply:
the rook cannot do it, the external-provisioner will get the secret data from the kubernetes based on the name and namespace and passes the secret data to the ceph-csi container